databricks/koalas/frame.py

```diff
@@ -10089,7 +10089,7 @@ def __getitem__(self, key):
 
         if key is None:
             raise KeyError("none key")
-        if isinstance(key, (str, tuple, list)):
+        if isinstance(key, (str, tuple, list, pd.Index)):
```

Comment:
Actually, I think it's okay to just change to `if isinstance(key, (str)) or is_list_like(key):` and `key = list(key) if is_list_like(key) else key` for simplicity for now.

Reply:
I think calling `list(key)` when `key` is a koalas series will generate a `databricks.koalas.exceptions.PandasNotImplementedError`. So something like `kdf[kdf["b"] > 2]` will fail.

Reply:
We can switch the if-else conditions. But okay, let's just keep it as is for now.

Reply:
Of course, makes sense. I've switched the conditions.